src/GraphBLAS-sharp/Abstracts.fs

```diff
@@ -12,56 +14,48 @@ type Matrix<'a when 'a : struct and 'a : equality>() =
     abstract Item: Mask1D option * int -> Vector<'a> with get, set
     abstract Item: int * Mask1D option -> Vector<'a> with get, set
     abstract Item: int * int -> Scalar<'a> with get, set
-    abstract Item: Mask2D option -> Scalar<'a> with set
-    abstract Item: Mask1D option * int -> Scalar<'a> with set
-    abstract Item: int * Mask1D option -> Scalar<'a> with set
-
-    abstract Mxm: Matrix<'a> -> Mask2D option -> Semiring<'a> -> Matrix<'a>
-    abstract Mxv: Vector<'a> -> Mask1D option -> Semiring<'a> -> Vector<'a>
-    abstract EWiseAdd: Matrix<'a> -> Mask2D option -> Semiring<'a> -> Matrix<'a>
-    abstract EWiseMult: Matrix<'a> -> Mask2D option -> Semiring<'a> -> Matrix<'a>
+    abstract Fill: Mask2D option -> Scalar<'a> with set
+    abstract Fill: Mask1D option * int -> Scalar<'a> with set
+    abstract Fill: int * Mask1D option -> Scalar<'a> with set
+
+    abstract Mxm: Matrix<'b> -> Mask2D option -> Semiring<'a, 'b, 'c> -> Matrix<'c>
+    abstract Mxv: Vector<'b> -> Mask1D option -> Semiring<'a, 'b, 'c> -> Vector<'c>
+    abstract EWiseAdd: Matrix<'a> -> Mask2D option -> BinaryOp<'a, 'b, 'c> -> Matrix<'a>
```

Comment:
Yes, it is necessary because there is a parent BFS algorithm, which use non-closed minFirst semiring. The alternative, which use embedding, doesn't seem very intuitive to me. Now, I can make the semiring closed, but this problem will require a more elegant solution in the future.